app/Controller/AppController.php

```diff
@@ -406,6 +406,12 @@ public function beforeFilter()
                     if (!$this->_isRest()) {
                         $this->redirect(array('controller' => 'users', 'action' => 'change_pw', 'admin' => false));
                     }
+                } elseif (Configure::read('Security.email_otp_enabled') && !$this->_isRest() && !in_array($this->request->here, array($base_dir.'/users/terms', $base_dir.'/users/email_otp', $base_dir.'/users/change_pw', $base_dir.'/users/logout', $base_dir.'/users/login'))) {
```

Comment:
need to find another way than `$this->_isRest()` to distinguish REST API users. This can be easily circumvented by adding some request header accept response types.  
We might need to review the location of this code. Authentication is already succeeded, and afterwards we do extra authentication elements.